tests/unittest_observer/FaceliftObserverTest.cpp

```diff
@@ -0,0 +1,52 @@
+#include <gtest/gtest.h>
+#include "IPCProxyBase.h"
+#include "InterfaceBase.h"
+
+namespace {
+
+using namespace facelift;
+
+class Counter
+{
+public:
+    Counter() { m_value = 0; }
+
+    void incValue() {
+        m_value++;
+	}
+    int getValue() {
+        return m_value;
+	}
+private:
+    int m_value;
+};
+
+class IPCProxyBaseTest : public ::testing::Test
+{
+public:
+	Counter c1;
+	Counter c2;
+	StandartObserver  < std::function<void()> > * obs1 = new StandartObserver  < std::function<void()> >(std::bind(&Counter::incValue, &c1) );
+	SingleTimeObserver< std::function<int()> > * obs2 = new SingleTimeObserver< std::function<int()> >(std::bind(&Counter::getValue, &c2) );
+
+    IPCProxyBase<InterfaceBase> proxyBase{nullptr};
+
+    ~IPCProxyBaseTest(){
+        delete obs1;
+        obs1 = nullptr;
+        delete obs2;
+        obs2 = nullptr;
+    }
+};
+
+TEST_F(IPCProxyBaseTest, setObservers)
+{
+    const auto expected = std::vector<IObserver*>{obs1, obs2,};
+    proxyBase.m_readyObserver.setObservers(expected);
+    const auto actual = proxyBase.m_readyObserver.getObservers();
+    EXPECT_EQ(expected.size(), actual.size());
+    EXPECT_EQ(expected[0], actual[0]);
+    EXPECT_EQ(expected[1], actual[1]);
+}
+
```

Comment:
What do you test?
Could you test the actual usage of new functionality?

Reply:
This is not a complete test cases. I will inform you. 

Wait, please.